general/device.hpp

```diff
@@ -61,31 +61,35 @@ struct Backend
           backend. Enabled when MFEM_USE_CEED = YES and MFEM_USE_CUDA = YES.
           NOTE: The current default libCEED GPU backend is non-deterministic! */
```

Comment:
`GPU` -> `CUDA` in this comment.

Reply:
Good catch, I updated it in aa278f92a3912fd5a648ed470c84f70668b708f8. 
I also removed the non-deterministic comment for the libCEED HIP backend, since due to recent changes in libCEED, the default HIP backend *is* deterministic now.